marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/savedQueries/saveSavedQuery.sjs

```diff
@@ -0,0 +1,56 @@
+'use strict';
+declareUpdate();
+
+var saveQuery;
+var userCollections = ["http://marklogic.com/data-hub/saved-query"];
+var queryDocument = JSON.parse(saveQuery);
+
+if (queryDocument == null || queryDocument.savedQuery == null) {
+    throw Error("The request is empty or malformed");
+}
+
+if (queryDocument.savedQuery.name == null || !queryDocument.savedQuery.name) {
+    throw Error("Query name is missing");
+}
+
+if (queryDocument.savedQuery.query == null || Object.keys(queryDocument.savedQuery.query) == 0) {
+    throw Error("Query to be saved cannot be empty");
+}
+
+if (queryDocument.savedQuery.propertiesToDisplay == null || queryDocument.savedQuery.propertiesToDisplay.length == 0) {
+    throw Error("Entity type properties to be displayed cannot be empty");
+}
+
+let id = queryDocument.savedQuery.id;
+
+if(id) {
+    if(cts.doc("/saved-queries/"+id+".json")) {
+        queryDocument.savedQuery.systemMetadata.lastUpdatedBy = xdmp.getCurrentUser();
+        queryDocument.savedQuery.systemMetadata.lastUpdatedDateTime = fn.currentDateTime();
+        xdmp.nodeReplace(cts.doc("/saved-queries/" + id+".json"), queryDocument);
+    } else {
+        throw Error("The requested resource with " + id + " is not found")
```

Comment:
I don't think this is necessary. I think you just need to say that if /saved-queries/id.json exists already, then nodeReplace it, else documentInsert it.